src/app/user/painting/component/painting-list/painting-list.component.ts

```diff
@@ -0,0 +1,93 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {PaintingListItem} from '../../../entity/painting-list/painting-list-item';
+import {PaintingService} from '../../service/painting.service';
+import {PaintingFilterService} from '../../filter/painting-filter.service';
+
+@Component({
+  selector: 'app-painting-list',
+  templateUrl: './painting-list.component.html',
+  styleUrls: ['./painting-list.component.scss']
+})
+export class PaintingListComponent implements OnInit {
+  public artists: string[];
+  public artTypes: string[];
+  paintingList: PaintingListItem[];
+  @Input() filter = true;
+
+  config: any;
+
+  filterActiveArtist: string = null;
+  filterActiveArtType: string = null;
+
+  paintingsView: {
+    id: number,
+    viewNumber: number
+  }[] = [];
+  paintingsLove: {
+    id: number,
+    loveNumber: number
+  }[] = [];
+
+  filterService: PaintingFilterService = null;
+
+  constructor(private paintingService: PaintingService) {
+  }
+
+  ngOnInit() {
+    this.paintingService.getPaintings().subscribe(
+      paintingList => {
+        this.paintingList = paintingList;
+        this.filterService = new PaintingFilterService(paintingList);
```

Comment:
as I said before , not allowed

Reply:
OK, moving to the second approach of using getters and setters